src/Core/Prefab/Panorama/PanoramaTileBuilder.js

```diff
@@ -97,57 +102,55 @@ PanoramaTileBuilder.prototype.vProjecte = function vProjecte(v, params) {
-        this.vProjecte(0, params);
-        pts.push(this.VertexPosition(params).clone());
-        this.uProjecte(1, params);
-        this.vProjecte(1, params);
-        pts.push(this.VertexPosition(params).clone());
-
-        const direction = params.center.clone();
-        direction.z = 0;
-        direction.normalize();
-
-        const diffExtent = new THREE.Vector3().subVectors(pts[2], pts[1]);
-        const height = diffExtent.z;
-        diffExtent.z = 0;
-
-        const length = diffExtent.length();
-
-        const diff = new THREE.Vector3().subVectors(params.center, pts[1]);
-        diff.z = 0;
-        const thickness = diff.dot(direction);
-
-        const min = new THREE.Vector3(-length * 0.5, -height * 0.5, -thickness * 0.5);
-        const max = new THREE.Vector3(length * 0.5, height * 0.5, thickness * 0.5);
-
-        const translate = new THREE.Vector3(0, 0, thickness * -0.5);
-        return new OBB(min, max, direction, translate);
-    }
+PanoramaTileBuilder.prototype.OBB = function _OBB(boundingBox) {
+    return new OBB(boundingBox.min, boundingBox.max);
+};
+
+PanoramaTileBuilder.prototype.getCommonGeometryExtent = function getCommonGeometryExtentFn(extent) {
+    const sizeLongitude = Math.abs(extent.west() - extent.east()) / 2;
+    const communExtent = new Extent(extent.crs(), -sizeLongitude, sizeLongitude, extent.south(), extent.north());
+    communExtent._internalStorageUnit = extent._internalStorageUnit;
+    return communExtent;
+};
+
+const axisY = new THREE.Vector3(0, 1, 0);
+const axisZ = new THREE.Vector3(0, 0, 1);
+const quatToAlignLongitude = new THREE.Quaternion();
+const quatToAlignLatitude = new THREE.Quaternion();
+
+PanoramaTileBuilder.prototype.getQuaternionFromExtent = function getQuaternionFromExtentFn(extentFrom, extentTo) {
+    const rotLon = extentTo.west(UNIT.RADIAN) - extentFrom.west(UNIT.RADIAN);
+    const rotLat = Math.PI * 0.5 - (!this.equirectangular ? 0 : (extentFrom.north(UNIT.RADIAN) + extentFrom.south(UNIT.RADIAN)) * 0.5);
+    quatToAlignLongitude.setFromAxisAngle(axisZ, -rotLon);
+    quatToAlignLatitude.setFromAxisAngle(axisY, -rotLat);
+    return quatToAlignLongitude.multiply(quatToAlignLatitude);
+};
+
+PanoramaTileBuilder.prototype.computeSharableExtent = function fnComputeSharableExtent(extent) {
+    // Compute sharable extent to pool the geometries
+    // the geometry in common extent is identical to the existing input
+    // with a transformation (translation, rotation)
+
+    // TODO: It should be possible to use equatorial plan symetrie,
+    // but we should be reverse UV on tile
+    // Common geometry is looking for only on longitude
+    const sizeLongitude = Math.abs(extent.west() - extent.east()) / 2;
+    const sharableExtent = new Extent(extent.crs(), -sizeLongitude, sizeLongitude, extent.south(), extent.north());
+    sharableExtent._internalStorageUnit = extent._internalStorageUnit;
+
+    // compute rotation to transform tile to position it on ellipsoid
```

Comment:
The comment refers to an "ellipsoid" but there's no ellipsoid in this context.
Can you please fix the comment?